entrypoint.sh:wait for mod to finish building before testing existence

andy5995 opened the pull request:
Also add a test

fixes #5

andy5995 committed:
entrypoint.sh:wait for mod to finish building before testing existence

Also add a test

fixes #5

andy5995 committed:
test @${{ github.head_ref }}

andy5995 committed:
temporarily hard-code branch for testing

andy5995 commented on .github/workflows/test.yml:
```diff
@@ -29,14 +27,23 @@ jobs:
       with:
         submodules: true
-    - uses: 0ad-matters/gh-action-build-pyromod@trunk
+    - uses: 0ad-matters/gh-action-build-pyromod@delay
```
Trying to find out how to set the correct branch name for this...

https://github.com/orgs/community/discussions/35812

StanleySweet commented:
I wondering why it didn't wait in the first place. Maybe because it's ran as another user?

andy5995 committed:
use action from same repository

Using an action in the same repository as the workflow

https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#example-using-an-action-in-the-same-repository-as-the-workflow

andy5995 commented:
> I wondering why it didn't wait in the first place. Maybe because it's ran as another user?

Looks like you're using a slightly outdated version of the example.

Note in https://github.com/0ADMods/xiiiad/actions/runs/3224966299/jobs/5276698900#step:5:28 it's using github.repository for the output filename. That would be 0ADMods/xiiiad. In prior examples, the example cut the leading 'owner/' from the MOD_NAME variable.

I think I've updated the examples everywhere, but maybe I'm wrong. Where did you get the one you're using from?

andy5995 commented:
@StanleySweet When I run this locally all as one command, there's no error

`{ ~/Downloads/0ad-0.0.26-alpha-2210062307-x86_64.AppImage      -mod=package_mod       -archivebuild=xiiiad      -archivebuild-output=xiiiad-test.pyromod        -archivebuild-compress;zip -d xiiiad-test.pyromod ".git*"; }`

StanleySweet commented:
I went to look for it  in the history because I wanted it to be dynamic in case I ever renamed the repo. Sucks that it includes the organisation name :/

andy5995 commented:
https://github.com/0ad-matters/gh-action-build-pyromod/commit/0a542da3dd05282fd49f209041a26c1e904d4d33#diff-b335630551682c19a781afebcf4d07bf978fb1f8ac04c6bf87428ed5106870f5R58

andy5995 commented:
> [0a542da#diff-b335630551682c19a781afebcf4d07bf978fb1f8ac04c6bf87428ed5106870f5R58](https://github.com/0ad-matters/gh-action-build-pyromod/commit/0a542da3dd05282fd49f209041a26c1e904d4d33#diff-b335630551682c19a781afebcf4d07bf978fb1f8ac04c6bf87428ed5106870f5R58)

@StanleySweet Another way to get the mod name dynamically is use the command-line json parser (not available by default on all Linux distros)

`MOD_NAME="$(jq .name mod.json | tr -d \")"`
